src/popup/pages/AppsBrowser.vue

```diff
@@ -147,6 +149,7 @@ export default defineComponent({
     const customAppURL = ref('');
     const currentClientId = ref('');
     let shareWalletInfoInterval: any;
+    let lastUrlAddedToHistory = '';
```

Comment:
as far as `load` event can be triggered multiple times for the same url, to don't mark it as visited for each load